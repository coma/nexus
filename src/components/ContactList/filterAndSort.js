export default function filterAndSort(items, search) {
  const q = search.toLowerCase();

  return items
    .filter(
      contact =>
        contact.name.last.toLowerCase().startsWith(q) ||
        contact.name.first.toLowerCase().startsWith(q),
    )
    .sort((a, b) => a.name.first.localeCompare(b.name.first));
}
