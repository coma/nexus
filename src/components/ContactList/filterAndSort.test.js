import filterAndSort from './filterAndSort';

describe('The filterAndSort helper', () => {
  it('should filter by firts and last name and sort by firts name', () => {
    expect(
      filterAndSort(
        [
          { id: '3', name: { first: '3', last: 'h' } },
          { id: 'al', name: { first: 'Alejandro', last: 'foo' } },
          { id: 'a', name: { first: 'a', last: 'foo' } },
          { id: 'x', name: { first: 'X', last: 'foo' } },
          { id: 'z', name: { first: 'zAlax', last: 'foo' } },
          { id: 'f', name: { first: 'tooo', last: 'lalala' } },
        ],
        'aL',
      ),
    ).toMatchSnapshot();
  });
});
