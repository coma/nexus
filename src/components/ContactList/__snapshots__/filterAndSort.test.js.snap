// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The filterAndSort helper should filter by firts and last name and sort by firts name 1`] = `
Array [
  Object {
    "id": "al",
    "name": Object {
      "first": "Alejandro",
      "last": "foo",
    },
  },
]
`;
